packages/@vuepress/theme-default/src/client/components/SidebarItems.vue

```diff
@@ -1,8 +1,44 @@
 <script setup lang="ts">
 import SidebarItem from '@theme/SidebarItem.vue'
+import { watch } from 'vue'
+import { useRouter } from 'vue-router'
 import { useSidebarItems } from '../composables'
 
 const sidebarItems = useSidebarItems()
+
+const router = useRouter()
+
+watch(
+  () => router.currentRoute.value.hash,
```

Comment:
I am not sure, could you explain why are you using `router` instead of `route`?

Why don't you choose:

```
const route = useRoute();

watch(() => route.hash, ...
```

Reply:
Yes, you're right! I should use `useRoute` to replace `useRouter` and I have changed it.